sn_networking/src/event/kad.rs

```diff
@@ -369,92 +382,98 @@ impl SwarmDriver {
-                    1
+                    result_map.insert(
+                        record_content_hash,
+                        (peer_record.record.clone(), peer_list.clone()),
+                    );
+
+                    peer_list
                 };
 
+            let responded_peers = peer_list.len();
+
             let expected_answers = get_quorum_value(&cfg.get_quorum);
+            trace!("Expecting {expected_answers:?} answers to exceed {expected_get_range:?} for record {pretty_key:?} task {query_id:?}, received {responded_peers} so far");
+            // return error if the entry cannot be found
+            return Err(NetworkError::ReceivedKademliaEventDropped {
+                query_id,
+                event: format!("Accumulate Get Record of {pretty_key:?}"),
+            });
+        }
+        Ok(())
+    }
 
-            debug!("Expecting {expected_answers:?} answers for record {pretty_key:?} task {query_id:?}, received {responded_peers} so far");
+    /// Checks passed peers from a request and checks they are sufficiently spaced to
+    /// ensure we have searched enough of the network range as determined by our `get_range`
+    ///
+    /// We expect any conflicting records to have been reported prior ti this check,
+    /// so we assume we're returning unique records only.
+    fn have_we_have_searched_thoroughly_for_quorum(
+        expected_get_range: KBucketDistance,
+        searched_peers_list: &HashSet<PeerId>,
+        data_key_address: &NetworkAddress,
+        quorum: &Quorum,
+    ) -> bool {
+        warn!("Assessing search: range: {:?}, address: {data_key_address:?}, quorum required: {quorum:?}, peers_returned_count: {:?}", expected_get_range.ilog2(), searched_peers_list.len());
+        let is_sensitive_data = matches!(quorum, Quorum::All);
+
+        let required_quorum = get_quorum_value(quorum);
+
+        let met_quorum = searched_peers_list.len() >= required_quorum;
+
+        // we only enforce range if we have sensitive data...for data spends quorum::all
+        if met_quorum && !is_sensitive_data {
+            return true;
+        }
 
-            if responded_peers >= expected_answers {
-                if !cfg.expected_holders.is_empty() {
-                    debug!("For record {pretty_key:?} task {query_id:?}, fetch completed with non-responded expected holders {:?}", cfg.expected_holders);
-                }
-                let cfg = cfg.clone();
+        // get the farthest distance between peers in the response
+        let mut max_distance_to_data_from_responded_nodes = KBucketDistance::default();
 
-                // Remove the query task and consume the variables.
-                let (_key, senders, result_map, _) = entry.remove();
+        // iterate over peers and see if the distance to the data is greater than the get_range
+        // Fathest peer from the data that has returned it
```

Comment:
`Fathest peer from the data that has returned it`
this sentence can now be removed ?